app/src/main/res/layout/fragment_crime.xml

```diff
@@ -1,52 +1,109 @@
 <?xml version="1.0" encoding="utf-8"?>
-<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```

Comment:
You have some nice reading material [here](https://developer.android.com/training/improving-layouts/optimizing-layout) and [here ](https://developer.android.com/topic/performance/rendering/optimizing-view-hierarchies)on why flat layout hierarchies matter